azure-communication-ui/demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/CallCompositeManager.kt

```diff
@@ -16,158 +15,191 @@ import com.azure.android.communication.common.CommunicationTokenCredential
 import com.azure.android.communication.ui.calling.models.CallCompositeRemoteOptions
+import com.azure.android.communication.ui.calling.models.CallCompositeRoomLocator
 import com.azure.android.communication.ui.calling.models.CallCompositeSetupScreenViewData
+import com.azure.android.communication.ui.calling.models.CallCompositeStartCallOptions
+import com.azure.android.communication.ui.calling.models.CallCompositeTeamsMeetingLinkLocator
 import com.azure.android.communication.ui.calling.models.CallCompositeTelecomIntegration
 import com.azure.android.communication.ui.calling.models.CallCompositeTelecomOptions
 import com.azure.android.communication.ui.callingcompositedemoapp.IncomingCallActivity.Companion.DISPLAY_NAME
+import com.azure.android.communication.ui.callingcompositedemoapp.events.AudioSelectionEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.CallCompositeEventsHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.CallExitEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.CallStateEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.ErrorEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.IncomingCallEndEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.IncomingCallEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.PiPEventHandler
+import com.azure.android.communication.ui.callingcompositedemoapp.events.RemoteParticipantJoinedEventHandler
 import com.azure.android.communication.ui.callingcompositedemoapp.features.AdditionalFeatures
 import com.azure.android.communication.ui.callingcompositedemoapp.features.SettingsFeatures
 import com.azure.android.communication.ui.callingcompositedemoapp.telecom.TelecomConnectionManager
+import com.azure.android.communication.ui.callingcompositedemoapp.views.EndCompositeButtonView
 import com.google.android.gms.tasks.OnCompleteListener
 import com.google.firebase.messaging.FirebaseMessaging
-import com.microsoft.appcenter.utils.HandlerUtils.runOnUiThread
-
-class CallCompositeManager(private var applicationContext: Context?) : CallCompositeEvents {
-
-    private val telecomConnectionManager: TelecomConnectionManager? =
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            TelecomConnectionManager.getInstance(
-                applicationContext!!,
-                TelecomConnectionManager.PHONE_ACCOUNT_ID
-            )
-        } else {
-            null
-        }
-    private var incomingCallEvent: IncomingCallEvent? = null
-    private var incomingCallEndEvent: IncomingCallEndEvent? = null
-    private var callStateEventHandler: CallStateEventHandler? = null
+import kotlinx.coroutines.flow.MutableStateFlow
+import java.util.UUID
+
+class CallCompositeManager(private var applicationContext: Context?) : CallCompositeEventsHandler {
```

Comment:
assert and enforce if this is application,

Because you are passing activity in.

Reply:
you can take base context, but you need to 
applicationContext = suppliedContext.applicationContext to guarantee it's application context.